src/System.Private.CoreLib/shared/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -130,6 +130,8 @@ public static bool TryLoad(string libraryName, Assembly assembly, DllImportSearc
         /// <param name="handle">The native library handle to be freed</param>
         public static void Free(IntPtr handle)
         {
+            if (handle == IntPtr.Zero)
```

Comment:
Could you please delete the check from the unmanaged side if you are adding it here?

Replace this: https://github.com/dotnet/coreclr/blob/ed5dc831b09a0bfed76ddad684008bebc86ab2f0/src/vm/dllimport.cpp#L6525 with `_ASSERTE(handle != NULL);`

Reply:
Done.